util/tls/tls.go

```diff
@@ -0,0 +1,126 @@
+package tls
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"time"
+)
+
+type CertOptions struct {
+	// Hostnames and IPs to generate a certificate for
+	Hosts []string
+	// Name of organization in certificate
+	Organization string
+	// Creation date
+	ValidFrom time.Time
+	// Duration that certificate is valid for
+	ValidFor time.Duration
+}
+
+func pemBlockForKey(priv interface{}) *pem.Block {
+	switch k := priv.(type) {
+	case *ecdsa.PrivateKey:
+		b, err := x509.MarshalECPrivateKey(k)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to marshal ECDSA private key: %v", err)
+			os.Exit(2)
+		}
+		return &pem.Block{Type: "EC PRIVATE KEY", Bytes: b}
+	default:
+		return nil
+	}
+}
+
+func generate(opts CertOptions) ([]byte, crypto.PrivateKey, error) {
+	if len(opts.Hosts) == 0 {
+		return nil, nil, fmt.Errorf("hosts not supplied")
+	}
+
+	var err error
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
What ECSDA rather that RSA?

Reply:
ECSDA is faster, and uses smaller keys to maintain the same/better level of security than that of RSA. Which would require a much larger key and comes with a performance hit.

Reply:
I totally agree with @davidcollom. But please make the particular algorithm configurable and not hard-coded. To test your stuff, I suggest you support both P256 and P521. P256 is OK as the default.